File: python/api/pocapi/simple.py
from __future__ import annotations

import requests

from pocapi.settings import POCAPI_URL


def avg_list(values: list) -> float | int | None:
    url = f"{POCAPI_URL}/simple/avg-list"
    response = requests.post(url, json={"values": values})
    return response.json()

File: r/api/src/simple.R
POCAPI_URL <- Sys.getenv("POCAPI_URL", unset="http://localhost:6666")

library(httr)

avg_list <- function(values){
    url <- paste(POCAPI_URL, "/simple/avg-list", sep="")
    r <- POST(url, body = list(values = values), encode = "json")
    return(content(r, as="parsed"))
}
